### PR TITLE
fix: better performance for labelled matrix display

### PR DESCRIPTION
**Pull request by thofma**
Current master:
```
julia> t = character_table("B");  # if you really want to see the formatted output then omit the semicolon

julia> @time show(IOBuffer(), "text/plain", t);
  9.962639 seconds (74.39 M allocations: 4.077 GiB, 5.35% gc time, 76.55% compilation time)

julia> @time show(IOBuffer(), "text/plain", t);
  2.275388 seconds (20.78 M allocations: 1.275 GiB, 3.93% gc time)
```
With this PR:
```
julia> t = character_table("B");  # if you really want to see the formatted output then omit the semicolon

julia> @time show(IOBuffer(), "text/plain", t);
  2.590985 seconds (8.90 M allocations: 575.139 MiB, 10.43% gc time, 95.47% compilation time)

julia> @time show(IOBuffer(), "text/plain", t);
  0.130060 seconds (1.57 M allocations: 76.671 MiB)
```

### CODE REVIEW

File: src/Groups/MatrixDisplay.jl [hunk 1]
```diff
@@ -250,21 +250,24 @@ function labeled_matrix_formatted(io::IO, mat::Matrix{String})
...
+      rightpart = map(x -> replace_TeX(x, is_unicode_allowed = with_unicode), rightpart)
 
       header = map(replace_TeX, header)
       footer = map(replace_TeX, footer)
 
       # ... compute columns widths of the corner entries and row labels ...
       leftwidth = map(textwidth, leftpart)
       leftrows = [leftwidth[i,:] for i in 1:m]
-      widths_leftpart = map(max, leftrows...)
+      #widths_leftpart = map(max, leftrows...)
```
Comment (ThomasBreuer) [inline src/Groups/MatrixDisplay.jl]: can be removed

**Review by ThomasBreuer — APPROVED** [review]
Nice, thanks.

File: src/Groups/MatrixDisplay.jl (hunk 1 above)
Comment (thofma) [inline src/Groups/MatrixDisplay.jl]:
```suggestion
```

File: src/Groups/MatrixDisplay.jl [hunk 2]
```diff
@@ -250,21 +250,23 @@ function labeled_matrix_formatted(io::IO, mat::Matrix{String})
...
 
       header = map(replace_TeX, header)
       footer = map(replace_TeX, footer)
 
       # ... compute columns widths of the corner entries and row labels ...
       leftwidth = map(textwidth, leftpart)
       leftrows = [leftwidth[i,:] for i in 1:m]
-      widths_leftpart = map(max, leftrows...)
+      widths_leftpart = [maximum(f[i] for f in leftrows; init = 0) for i in 1:length(first(leftrows))]
+
```
Comment (thofma) [inline src/Groups/MatrixDisplay.jl:264]:
```suggestion
```

**Pull request merged**